zebra-network/src/address_book.rs

```diff
@@ -72,6 +73,14 @@ pub struct AddressBook {
     /// [`OrderedMap`] sorts in descending order.
     by_addr: OrderedMap<PeerSocketAddr, MetaAddr, Reverse<MetaAddr>>,
 
+    /// The address with a last_connection_state of [`PeerAddrState::Responded`] and
+    /// the most recent `last_response` time by IP.
+    ///
+    /// This is used to avoid initiating outbound connections past [`Config::max_connections_per_ip`](crate::config::Config), and
+    /// currently only supports a `max_connections_per_ip` of 1, and must be `None` when used with a greater `max_connections_per_ip`.
+    // TODO: Replace with `by_ip` to support configured `max_connections_per_ip`
```

Comment:
This is a potential design for another ticket, which would keep the complexity to `O(log(n))`.

```suggestion
    // TODO: Replace with `by_ip: BTreeMap<(DateTime32, IpAddr), MetaAddr>` and use binary search to support configured `max_connections_per_ip`
```